src/libostree/ostree-sepolicy.c

```diff
@@ -163,80 +163,85 @@ ostree_sepolicy_class_init (OstreeSePolicyClass *klass)
 
 /* Find the latest policy file in our root and return its checksum. */
 static gboolean
-get_policy_checksum (char **out_csum,
+get_policy_checksum (char        **out_csum,
                      GCancellable *cancellable,
-                     GError **error)
+                     GError      **error)
 {
   gboolean ret = FALSE;
+
   const char *binary_policy_path = selinux_binary_policy_path ();
   g_autofree char *bindir_path = g_path_get_dirname (binary_policy_path);
-  g_autoptr(GFile) bindir = g_file_new_for_path (bindir_path);
-  g_autoptr(GFile) best_child = NULL;
+  g_autofree char *binfile_prefix = g_path_get_basename (binary_policy_path);
+
+  glnx_fd_close int bindir_dfd = -1;
+
+  g_autofree char *best_policy = NULL;
   int best_version = 0;
 
-  /* The binary_policy_path includes the prefix of the basename, but without the
-   * period, e.g. "policy". */
-  g_autofree char *binfile_prefix = g_path_get_basename (binary_policy_path);
-  g_autofree char *binfile_no_fnmatch = g_strdup_printf ("%s.*[!0-9]*",
-                                                         binfile_prefix);
+  static gsize regex_initialized;
+  static GRegex *regex;
 
-  g_autoptr(GFileEnumerator) direnum =
-    g_file_enumerate_children (bindir, OSTREE_GIO_FAST_QUERYINFO,
-                               G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS,
-                               cancellable, error);
+  g_auto(GLnxDirFdIterator) dfd_iter = { 0,};
 
-  if (!direnum)
+  if (g_once_init_enter (&regex_initialized))
+    {
+      g_autofree char *sregex = g_strdup_printf ("^\\Q%s\\E\\.[0-9]+$",
+                                                 binfile_prefix);
```

Comment:
No I agree. I was coding everything else as if the prefix could be different from `policy`, so might as well go all the way here and make it per-instantiation. Since this is called from the initable_init, we could just leave it here and just not make it static, no? (Or maybe just use `g_regex_match_simple()` ?)